src/managers/UpdateManager.ts

```diff
@@ -23,6 +23,10 @@ export class UpdateManager {
     this.onSessionSent = context.pageViewManager.getPageViewCount() > 1;
   }
 
+  private async isDeviceIdAvailable(): Promise<boolean> {
+    return (await Database.getSubscription()).deviceId != null;
```

Comment:
when is this set to `null`? do you mean `undefined`?

or are you purposefully using `!=`  ?

Reply:
yes, `deviceId` maybe `undefined` or `null` based on the return type we defined.